src/expr/subs.cpp

```diff
@@ -0,0 +1,177 @@
+/*********************                                                        */
+/*! \file subs.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Simple substitution utility
+ **/
+
+#include "expr/subs.h"
+
+#include "theory/rewriter.h"
+
+namespace CVC4 {
```

Comment:
should this be only in the CVC4 namespace or in the `expr` namespace too?

Reply:
Many things in src/expr/ are intentionally only in the CVC4:: namespace.  This goes against the policy of namespaces matching folder directories, but is intentional since it is less verbose.  This utility in particular is preferred to be used as `Subs` instead of `expr::Subs`.